### PR TITLE
Include-What-You-Use: first pass of the files [WIP]

### PR DESCRIPTION
**Pull request by madebr**
In this PR:
- fix collecting coverage twice in `experimental_script.ctest` 
- first pass of the files using [include-what-you-use](https://include-what-you-use.org/):
  * I did rwlib and rwengine.
  * I still have rwgame and the tests to do (depending on your review)
  * There still are warnings. The mapping file should probably be extended more.
  * As a side effect of this first (partial) pass, the compile time (using `ninja`) on my system decreased from 1m21s to 1m14s.
    * The build times also shortened on Travis, when looking at CDash (by 20s).
    * e.g. compare [before](http://my.cdash.org/buildSummary.php?buildid=1315003) and [after](http://my.cdash.org/buildSummary.php?buildid=1315599):
      - before: 2017-10-31T01:20:52 EDT -> 2017-10-31T01:25:16 EDT = 4m24s
      - after: 2017-11-01T00:21:42 EDT -> 2017-11-01T00:25:48 EDT = 4m6s
- ~I converted the tabs to spaces in `rwengine/src/script/modules/GTA3ModuleImpl.inl`~ (See #366)
- ~I also added a patch to let the tests compile on newer boost versions (as mentioned on IRC)~

I would love some remarks about this effort. Is it worthwhile? What should be done otherwise?

### CODE REVIEW

File: rwengine/src/ai/TrafficDirector.hpp [hunk 1]
```diff
@@ -1,12 +1,11 @@
-#pragma once
+#ifndef _TRAFFICDIRECTOR_HPP_
+#define _TRAFFICDIRECTOR_HPP_
```
Comment (danhedron) [inline rwengine/src/ai/TrafficDirector.hpp]: This one is missing the module name

File: rwengine/src/audio/SoundManager.hpp [hunk 2]
```diff
@@ -1,12 +1,15 @@
-#pragma once
-
-#include "al.h"
-#include "alc.h"
+#ifndef _SOUNDMANAGER_HPP_
+#define _SOUNDMANAGER_HPP_
```
Comment (danhedron) [inline rwengine/src/audio/SoundManager.hpp]: here too

File: rwengine/src/audio/alCheck.hpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
-#pragma once
+#ifndef _ALCHECK_HPP_
+#define _ALCHECK_HPP_
```
Comment (danhedron) [inline rwengine/src/audio/alCheck.hpp]: here too

File: rwengine/src/render/DebugDraw.hpp [hunk 4]
```diff
@@ -1,8 +1,21 @@
-#pragma once
+#ifndef _DEBUGDRAW_HPP_
+#define _DEBUGDRAW_HPP_
```
Comment (danhedron) [inline rwengine/src/render/DebugDraw.hpp]: here too

File: rwengine/src/render/MapRenderer.hpp [hunk 5]
```diff
@@ -1,6 +1,16 @@
-#pragma once
+#ifndef _MAPRENDERER_HPP_
+#define _MAPRENDERER_HPP_
```
Comment (danhedron) [inline rwengine/src/render/MapRenderer.hpp]: here too

File: rwengine/src/render/VisualFX.hpp [hunk 6]
```diff
@@ -1,8 +1,10 @@
-#pragma once
+#ifndef _VISUALFX_HPP_
+#define _VISUALFX_HPP_
```
Comment (danhedron) [inline rwengine/src/render/VisualFX.hpp]: here too

File: rwengine/src/render/WaterRenderer.hpp [hunk 7]
```diff
@@ -1,7 +1,15 @@
-#pragma once
+#ifndef _WATERRENDERER_HPP_
+#define _WATERRENDERER_HPP_
```
Comment (danhedron) [inline rwengine/src/render/WaterRenderer.hpp]: here too

File: rwlib/source/gl/TextureData.hpp [hunk 8]
```diff
@@ -1,9 +1,11 @@
-#pragma once
+#ifndef _TEXTUREDATA_HPP_
+#define _TEXTUREDATA_HPP_
```
Comment (danhedron) [inline rwlib/source/gl/TextureData.hpp]: here too

File: rwengine/src/ai/TrafficDirector.hpp (hunk 1 above)
Comment (madebr) [reply rwengine/src/ai/TrafficDirector.hpp]: Ok, did not look at that. I'll use `_{MODULE}_{FILENAME}_HPP_`

**Comment by madebr** [comment]
@danhedron , I believe I have modified all Include Guards to include the module (librw or rwengine)

**Comment by danhedron** [comment]
Could you do the re-formatting of GTA3ModuleImpl.inl in another PR? It's quite a large diff so I'd like to merge it separately.

File: rwengine/src/data/PathData.hpp [hunk 9]
```diff
@@ -1,6 +1,5 @@
-#pragma once
-#ifndef __GLT_PATHDATA_HPP__
-#define __GLT_PATHDATA_HPP__
+#ifndef _RWENGIEN_PATHDATA_HPP_
```
Comment (danhedron) [inline rwengine/src/data/PathData.hpp]: typo

Comment (madebr) [reply rwengine/src/data/PathData.hpp]: ~FXIED~ -> FIXED

**Comment by danhedron** [comment]
LGTM.

**Pull request merged**